docs/electrical/embedded/styleguide.md

```diff
@@ -1,3 +1,136 @@
 
-*Note that this page is currently an outline and most of the content is not yet written.*
+*Note that this page is currently in an unfinished state and in the process of being written*
 
+## Overview
+This guide follows the general outline and some of the recommendations of [*Barr Group's Embedded C Coding Standard*](https://barrgroup.com/embedded-systems/books/embedded-c-coding-standard), set forth by Michael Barr.
+
+Many of these rules are common sense and you probably are already following them in your own coding style. However, for consistency across the team it is practical to write them explicitly.
+
+??? info "Revision History"
+    - *Revision 0.1 by William K, 11 Jun 2023*
+    - *Revision 0.2 by William K, 12 Aug 2023*
+
+---
+
+## Rules
+
+### General Rules
+#### Line Width
+1. No line width is enforced as we do not do paper code reviews. Please use your best descretion for when a line is considered too long.
+2. The preprocessor directive (```#define```) shall not be used to alter or rename any keyword or other part of the C programming language.
+3. The minimum target standard for both C and C++ is **C/C++ 11**. However, if newer versions such as 14 or 17 are available for the toolchain in use, they are preferred.
+   
+#### Braces
+1. Left braces should appear on the same line as the statement they are connected with. Right braces should always be on the line after the line of code in the block if the block is more than one line.
+   
+    Should the block be only one line long for a *conditional*, the entire statement including the right brace should be placed on the same line
```

Comment:
I discourage this. IMO, contents of a block should always be on a different line. Makes the code more readable especially for less experienced developers.